File: contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        if (k == 0 || nums.size() <= 1) return false;
        map<int,int>s;
        int tail = 0;
        for (int i = 0; i < nums.size(); ++i) {
            if (i - tail > k) {
                s[nums[tail]] = 0;
                tail++;
            }
            if (s[nums[i]] == 1) {
                return true;
            } else {
                s[nums[i]] = 1;
            }
        }
        return false;
    }
};

File: contains-duplicate-ii.py
class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        dic = dict();
        l = len(nums);
        for i in range(l):
            index = dic.get(nums[i])
            if index >= 0 and i - index <= k:
                return True;
            dic[nums[i]] = i;
        return False;